sample-plugin-bar-chart/src/App.js

```diff
@@ -0,0 +1,148 @@
+import {
+  client,
+  useConfig,
+  useElementData,
+  useElementColumns,
+} from "@sigmacomputing/plugin";
+import React from "react";
+import {
+  ResponsiveContainer,
+  BarChart,
+  Bar,
+  XAxis,
+  YAxis,
+  CartesianGrid,
+  Tooltip,
+  Legend,
+} from "recharts";
+
+client.config.configureEditorPanel([
+  {
+    // configuring selection of data source
+    name: "source",
+    type: "element",
+  },
+  {
+    // configuring selection of data column from the source that contains qualitative values
+    // (e.g. names, dates)
+    name: "dimension",
+    type: "column",
+    source: "source",
+    allowMultiple: false,
+  },
+  {
+    // configuring selection of data column(s) from the source that contain quantitative, numeric values
+    name: "measures",
+    type: "column",
+    source: "source",
+    allowMultiple: true,
+  },
+]);
+
+// chart layout via Recharts
+function RenderBarChart(props) {
+  return (
+    <ResponsiveContainer width="100%" height={500}>
+      <BarChart
+        data={props.data}
+        margin={{
+          top: 0,
+          right: 50,
+          left: 30,
+          bottom: 90,
+        }}
+        barCategoryGap={5}
+      >
+        <CartesianGrid strokeDasharray="3 3" />
+        <XAxis
+          dataKey={props.xAxisDataKey}
+          angle={-45}
+          textAnchor="end"
+          interval={0}
+        />
+        <YAxis />
+        <Tooltip />
+        <Legend verticalAlign="top" wrapperStyle={{ lineHeight: "40px" }} />
+        {props.measureNames.slice(0, 3).map(
+          // limiting number of bars rendered to max 3
+          (m, i) => (
+            <Bar
+              key={m}
+              dataKey={props.measureNames[i]}
+              fill={props.barFills[i]}
+            />
+          )
+        )}
+      </BarChart>
+    </ResponsiveContainer>
+  );
+}
+
+function App() {
+  const config = useConfig();
+
+  // object of arrays, where each array contains data from the columns of the
+  // data source selected in the editor panel
```

Comment:
We might want to try a broader description at a higher-level and remove some of the individual comments. Maybe we start with something like

Sigma represents their data in a columnar format. Given a data source, useElementData will return any columns specified in the side panel with their respective data. The format is { [columnId: string]: any[] }

For more details / examples on the API, go here: (link to Diana's docs)

Reply:
Unfortunately I google doc is private, and we do not have public hosted documentation anywhere. Although people who land here should have received the latest beta docs in form of PDF